service/tbc/notifier.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2025 Hemi Labs, Inc.
+// Use of this source code is governed by the MIT License,
+// which can be found in the LICENSE file.
+
+package tbc
+
+import (
+	"context"
+	"crypto/rand"
+	"io"
+	"sync"
+)
+
+type notification string
+
+const (
+	NotificationBlock       notification = "block inserted"
+	NotificationBlockheader notification = "blockheader inserted"
+)
```

Comment:
This needs some thought. I do think we need a notification type that has a String that returns a human readable notification. But we do need context. Like `block inserted 4398578394579843578943579843`.

We can make this:
```
type Notification Struct {
  Type string // the const above
  ID string // Or chainhash or any?  string may be nice to leave interpretation to caller
  Timestamp time.Time // Always :-)
  Error error // nil when shit's good
}
```